pyrealm/core/solar.py

```diff
@@ -1,26 +1,584 @@
+
+    This function calculates the net radiation crossover hour angle in degrees.
+
+    Args:
+        rnl         : net longwave radiation, W/m^2
+        rw          : intermediate variable, W/m^2
+        ru          : intermediate variable, W/m^2
+        rv          : intermediate variable, W/m^2
+        k_pir       : conversion factor from radians to degrees
+
+    Returns:
+        hn          : net radiation crossover hour angle, degrees
+    """
+
+    hn = np.arccos(np.clip((rnl - rw * ru) / (rw * rv), -1.0, 1.0)) / k_pir
+
+    return hn
+
+
+def calc_daytime_net_radiation(
+    hn: NDArray,
+    k_pir: float,
+    rnl: NDArray,
+    delta: NDArray,
+    latitude: NDArray,
+    k_alb_sw: float,
+    tau: NDArray,
+    k_Gsc: float,
+    dr: NDArray,
+) -> NDArray:
+    r"""Calculates daily net radiation, J/m^2.
+
+    .. math::
+        secs\_d = 86400  \quad \text{ \# seconds in one solar day}
+
+        rn\_d = \left( \frac{secs\_d}{\pi} \right) \cdot 
+        \left( hn \cdot k\_pir \cdot (rw \cdot ru - rnl) + rw \cdot rv \cdot \
+        sin(\deg2rad(hn)) \right)
+
+    Args:
+        hn          : crossover hour angle, degrees
+        k_pir       : conversion factor from radians to degrees
+        rnl         : net longwave radiation, W/m^2
+        delta       : solar declination delta
+        latitude    : site latitude(s)
+        k_alb_sw    : shortwave albedo
+        tau         : bulk transmissivity, unitless
+        k_Gsc       : solar constant, W/m^2
+        dr          : distance ration, unitless
+
+    Result:
+        _calc_daytime_net_radiation
+    """
+    ru, rv = calc_lat_delta_intermediates(delta, latitude)
+    rw = calc_rw(k_alb_sw, tau, k_Gsc, dr)
+
+    return _calc_daytime_net_radiation(hn, k_pir, rw, ru, rv, rnl)
+
+
+def _calc_daytime_net_radiation(
+    hn: NDArray, k_pir: float, rw: NDArray, ru: NDArray, rv: NDArray, rnl: NDArray
+) -> NDArray:
+    """Calculates daily net radiation, J/m^2.
+
+    Args:
+        hn          : crossover hour angle, degrees
+        k_pir       : conversion factor from radians to degrees
+        rw          : dimensionless variable substitute
+        ru          : dimensionless variable substitute
+        rv          : dimensionless variable substitute
+        rnl         : net longwave radiation, W/m^2
+
+    Result:
+        rn_d        : daily net radiation, J/m^2
+    """
+
+    secs_d = 86400  # seconds in one solar day
+
+    rn_d = (secs_d / np.pi) * (
+        hn * k_pir * (rw * ru - rnl) + rw * rv * np.sin(np.deg2rad(hn))
+    )
+
+    return rn_d
+
+
+def calc_nighttime_net_radiation(
+    k_pir: float,
+    rnl: NDArray,
+    hn: NDArray,
+    hs: NDArray,
+    delta: NDArray,
+    latitude: NDArray,
+    k_alb_sw: float,
+    tau: NDArray,
+    k_Gsc: float,
+    dr: NDArray,
+) -> NDArray:
+    r"""Calculates nightime net radiation, J/m^2.
+
+    .. math::
+
+        secs\_d = 86400  \quad \text{ \# seconds in one solar day}
```

Comment:
@davidorme should we move seconds in one solar day into core constants?

Reply:
Happy to! Although it seems obvious, pinning it to a clear definition is better.